server/src/main/java/org/elasticsearch/action/support/ActiveShardsObserver.java

```diff
@@ -16,79 +16,74 @@
 import org.elasticsearch.cluster.service.ClusterService;
 import org.elasticsearch.core.TimeValue;
 import org.elasticsearch.node.NodeClosedException;
-import org.elasticsearch.threadpool.ThreadPool;
 
 import java.util.Arrays;
-import java.util.function.Consumer;
-import java.util.function.Predicate;
+
+import static org.elasticsearch.core.Strings.format;
 
 /**
- * This class provides primitives for waiting for a configured number of shards
+ * This utility class provides a primitive for waiting for a configured number of shards
  * to become active before sending a response on an {@link ActionListener}.
  */
-public class ActiveShardsObserver {
-
+public enum ActiveShardsObserver {
+    ;
```

Comment:
I don't know, eventually we started doing this here and there. I kind of like it since it saves 3 lines for the constructor so I started doing this left and right :)